### DISCUSSION TITLE
how to use auth or basicauth with a form or a static_fileserver

### DISCUSSION THREAD

**Discussion by omac777** [rwf2/Rocket#1833] [Questions]
I noticed a few crates for auth floating around:
https://crates.io/crates/rocket_auth/0.3.0
https://crates.io/crates/rocket-basicauth

Unfortunately, rocket_auth requires an sql database to work.  
I was hoping perhaps a mongodb rust driver examples existed with something like rocket_auth but I didn't find any.

Also unfortunately, I did not find an example showing how to use rocket-basicauth with the existing static_fileserver example.
I didn't find an example using a form with rocket-basicauth either.
I didn't want to get any databases involved just yet.  I preferred to wire those in myself after I felt comfortable just getting this functionality working.  Could anybody point me in the right direction for this?

Thank you in advance.

**Comment by jebrosen** [comment]
I am not sure what you are trying to accomplish: The two crates you listed do very, very different things from each other:

`rocket-basicauth` implements the `Authorization: Basic` method. Most web browsers support this method; the server sends a `WWW-Authenticate` header to request it, and the web browser shows a username/password entry. However, this authentication method has declined in popularity overall - presumably because the web browser, not the website, has full control over how the login prompt appears.

`rocket_auth` implements a few request guards to check a user's password, add and delete users, etc. from a database.

Which crate (if either) sounds like what you are trying to do?
